src/Yoga/JSON.purs

```diff
@@ -455,4 +459,22 @@ instance ReadForeign a ⇒ ReadForeign (NonEmptyArray a) where
       $ fromArray raw
 
 instance writeForeignNEArray ∷ WriteForeign a ⇒ WriteForeign (NonEmptyArray a) where
-  writeImpl a = writeImpl <<< toArray $ a
\ No newline at end of file
+  writeImpl a = writeImpl <<< toArray $ a
+
+instance (WriteForeign a) => WriteForeign (Map String a) where
+  writeImpl = (Map.toUnfoldable :: _ -> Array _) >>> Object.fromFoldable >>> writeImpl
+else
+instance (Newtype key String, WriteForeign a) => WriteForeign (Map key a) where
+  writeImpl = coerceKeysToString >>> (Map.toUnfoldable :: _ -> Array _) >>> Object.fromFoldable >>> writeImpl
+    where
+    coerceKeysToString :: Map key a -> Map String a
+    coerceKeysToString = unsafeCoerce
+
+instance (ReadForeign a) => ReadForeign (Map String a) where
+  readImpl = readImpl >>> map ((Object.toUnfoldable :: _ -> Array _) >>> Map.fromFoldable)
+else
+instance (Newtype key String, ReadForeign a) => ReadForeign (Map key a) where
+  readImpl = readImpl >>> map ((Object.toUnfoldable :: _ -> Array _) >>> Map.fromFoldable >>> coerceKeysFromString)
```

Comment:
@sigma-andex  do you know how to avoid the intermediate step here most optimally?

Reply:
I guess you have solved this?

Reply:
suboptimally yes